plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/GeneralUtils.java

```diff
@@ -798,15 +799,19 @@ public static URI makeURIFromFilePath(@NotNull String path) throws URISyntaxExce
         return new URI(path.replace(" ", "%20"));
     }
 
+    public static <T> T adapt(Object sourceObject, Class<T> adapter, boolean allowActivation) {
+        return adapt(sourceObject, adapter, allowActivation, adapter::isInstance);
+    }
+
     /////////////////////////////////////////////////////////////////////////
     // Adapters
     // Copy-pasted from org.eclipse.core.runtime.Adapters to support Eclipse Mars (#46667)
 
-    public static <T> T adapt(Object sourceObject, Class<T> adapter, boolean allowActivation) {
+    public static <T> T adapt(Object sourceObject, Class<T> adapter, boolean allowActivation, Predicate<Object> isInstance) {
```

Comment:
Could we call this parameter in a different way? Maybe castPredicate, castTest, ... (I suppose you can figure it out better, than me). Currently it's not clear what should I pass to the parameter with this name only be the method contract.